apps/automation/automation-v2/src/tlc/batch-liquidate.ts

```diff
@@ -96,78 +107,85 @@ export async function batchLiquidate(
   // e.g. try to liquidate 1000 accounts at once could use too much gas and fail
   const accListChunks = chunkify(accsToLiquidate, config.ACC_LIQ_MAX_CHUNK_NO);
   for (const accBatch of accListChunks) {
-    const tx = await tlc.batchLiquidate(accBatch, {
-      gasLimit: config.GAS_LIMIT,
+    const data = encodeFunctionData({
+      abi: TLC_ABI,
+      functionName: 'batchLiquidate',
+      args: [accBatch]
     });
-    const txReceipt = await tx.wait();
-    if (!txReceipt) throw Error('undefined tx receipt');
+    const tx = { data, to: config.TLC_ADDRESS };
+    const txr = await transactionManager.submitAndWait(tx);
+    if (!txr) throw Error('undefined tx receipt');
 
     // Grab the events
     const events: TempleTaskDiscordEvent[] = [];
 
-    await txReceipt.logs.forEach((log) => {
-      if (log instanceof EventLog) {
-        const liquidatedEv = matchAndDecodeEvent(
-          log,
-          tlc,
-          tlc.filters.Liquidated()
-        );
+    txr.logs.forEach((log) => {
+      try {
+        const decodedLog = decodeEventLog({
+          abi: TLC_ABI,
+          data: log.data,
+          topics: log.topics,
+        });
 
         // check if this is the liquidated event
-        if (liquidatedEv) {
+        if (decodedLog.eventName === 'Liquidated') {
+          const args = decodedLog.args as any;
           events.push({
             what: 'Liquidated',
             details: [
-              `account = \`${liquidatedEv.account}\``,
+              `account = \`${args.account}\``,
               `collateralValue = \`${formatBigNumber(
-                liquidatedEv.collateralValue,
+                args.collateralValue,
                 18,
                 4
               )}\``,
               `collateralSeized = \`${formatBigNumber(
-                liquidatedEv.collateralSeized,
+                args.collateralSeized,
                 18,
                 4
               )}\``,
               `daiDebtWiped = \`${formatBigNumber(
-                liquidatedEv.daiDebtWiped,
+                args.daiDebtWiped,
                 18,
                 4
               )}\``,
             ],
           });
         }
+      } catch (e: any) {
+        // Log what happend while decoding event log
+        ctx.logger.error(`Error while decoding event log: ${e}`);
```

Comment:
This should not fail silently, I would expect an error thrown at the end of the loop if there is any error. Do we expect a failure here though? I would be surprised to see one

Reply:
Good call, no error should be expected here.